Protocol.md

```diff
@@ -0,0 +1,177 @@
+# DLC Protocol
+
+This document describes a protocol for two parties to set up a Discreet Log Contract (DLC).
+
+## Preliminaries
+
+The following protocol assumes that both parties have already agreed on the contract terms (the possible outputs of the contract).
+Negotiation of the contract, or decisions on the underlying asset are thus not part of the exchanged information.
+
+## Description
+
+The two parties exchange a set of three messages, and terminates when the non-initiating party broadcasts the Fund Transaction to the Bitcoin network.
+The following figure gives a brief overview of the protocol:
+
+    +-------+                    +-------+
+    |       |                    |       |
+    |       |---- (1) offer  --->|       |
+    |       |                    |       |
+    |       |<--- (2) accept ----|       |
+    |   A   |                    |   B   |
+    |       |---- (3) sign   --->|       |
+    |       |                    |       |
+    |       |                    |  (4) broadcast fund-tx
+    |       |                    |       |
+    +-------+                    +-------+
+
+### Offer
+
+The initiating party starts the protocol by sending an `offer` message to the other party.
+This message includes the following information:
+1. Contract information (an array of triples defined below)
+1. Oracle information (optional)
+1. A's fund amount (satoshi)
+1. A's public key
```

Comment:
nit: extended public key (link to key generation)

Reply:
Updated